### DISCUSSION TITLE
How to send email notifications/alerts and send logs from a fortigate to security onion

### DISCUSSION THREAD

**Discussion by juanwoczinski** [Security-Onion-Solutions/securityonion#7966] [Unsupported Versions] [answered]
I have a Fortigate Firewall that was configured to send the logs through the SYSLOG port, the logs appear inside Kibana but do not appear in a structured way, how can I format these logs to improve the visualization?

Is there a possibility to send an email notification, when something in the log of this fortigate reaches some pre-configured metric?

**Below is an example of JSON that is displayed by Kibana:**

{
  "_index": "xxx:so-syslog-2022.05.17",
  "_type": "_doc",
  "_id": "DBae04ABprZCQL2JDBMp",
  "_version": 1,
  "_score": 1,
  "_source": {
    "metadata": {
      "pipeline": "syslog",
      "raw_index": "so-syslog",
      "beat": "filebeat",
      "truncated": false,
      "ip_address": "xxxxx",
      "type": "_doc",
      "version": "7.17.1"
    },
    "log": {
      "source": {
        "address": "xxxxxx
      }
    },
    "message": "<189>date=2022-05-17 time=17:03:11 devname=\"xxxxxx\" devid=\"xxxxxxxx\" logid=\"0000000013\" type=\"traffic\" subtype=\"forward\" level=\"notice\" vd=\"root\" eventtime=1652817791556949540 tz=\"-0300\" srcip=xxxx srcport=xxxx srcintf=\"lan5\" srcintfrole=\"lan \" dstip=8.8.8.8 dstport=xxxx dstintf=\"wan2\" dstintfrole=\"wan\" dstinetsvc=\"Google-DNS\" sessionid=xxxx proto=17 action=\"accept\" policyid=47 policytype =\"policy\" poluuid=\"bfbf128e-8853-51e9-4dba-00103de9d520\" dstcountry=\"United States\" srccountry=\"Reserved\" trandisp=\"snat\" transip=xxxxxtransport=60748 duration= 181 sentbyte=104 rcvdbyte=259 sentpkt=1 rcvdpkt=1 vwlid=0 appcat=\"unscanned\" mastersrcmac=\"xx:4f:xx:34:xx\" srcmac=\"xx:4f:64:34: xxx\" srcserver=0",
    "tags": [
      "beats_input_codec_plain_applied"
    ],
    "@timestamp": "2022-05-17T20:03:11.444Z",
    "ecs": {
      "version": "1.12.0"
    },
    "host": {
      "name": "sslsiem01"
    },
    "@version": "1",
    "event": {
      "ingested": "2022-05-17T20:03:11.444Z",
      "module": "syslog",
      "dataset": "syslog"
    }
  },
  "fields": {
    "@version.security": [
      "1"
    ],
    "metadata.beat.keyword": [
      "filebeat"
    ],
    "host.name.keyword": [
      "xxxxx"
    ],
    "event.dataset.keyword": [
      "syslog"
    ],
    "tags.keyword": [
      "beats_input_codec_plain_applied"
    ],
    "metadata.pipeline.security": [
      "syslog"
    ],
    "metadata.raw_index.keyword": [
      "so-syslog"
    ],
    "metadata.beat.security": [
      "filebeat"
    ],
    "message.security": [
      "<189>date=2022-05-17 time=17:03:11 devname=\"xxxx\" devid=\"xxxx\" logid=\"0000000013\" type=\"traffic\" subtype=\" forward\" level=\"notice\" vd=\"root\" eventtime=xxxxtz=\"-0300\" srcip=xxxxxxsrcport=60748 srcintf=\"lan5\" srcintfrole=\"lan\" dstip=xxxxxxdstport= 53 dstintf=\"wan2\" dstintfrole=\"wan\" dstinetsvc=\"Google-DNS\" sessionid=10438009 proto=17 action=\"accept\" policyid=47 policytype=\"policy\" polluuid=\ "bfbf128e-8853-51e9-4dba-00103de9d520\" dstcountry=\"United States\" srccountry=\"Reserved\" trandisp=\"snat\" transip=xxxxxxtransport=60748 duration=181 sentbyte=104 rcvdbyte=259 sentpkt= 1 rcvdpkt=1 vwlid=0 appcat=\"unscanned\" mastersrcmac=\"68:xx:64:x\" srcmac=\"68:xx:64:xxx\" srcserver=0"
    ],
    "ecs.version.keyword": [
      "1.12.0"
    ],
    "event.module": [
      "syslog"
    ],
    "metadata.pipeline.keyword": [
      "syslog"
    ],
    "@version": [
      "1"
    ],
    "event.ingested.keyword": [
      "2022-05-17T20:03:11.444Z"
    ],
    "host.name": [
      "xxxxxx"
    ],
    "metadata.raw_index": [
      "so-syslog"
    ],
    "metadata.type": [
      "_doc"
    ],
    "ecs.version.security": [
      "1.12.0"
    ],
    "metadata.beat": [
      "filebeat"
    ],
    "metadata.pipeline": [
      "syslog"
    ],
    "metadata.version": [
      "7.17.1"
    ],
    "metadata.truncated": [
      false
    ],
    "event.dataset.security": [
      "syslog"
    ],
    "@version.keyword": [
      "1"
    ],
    "metadata.version.keyword": [
      "7.17.1"
    ],
    "message": [
      "<189>date=2022-05-17 time=17:03:11 devname=\"xxxx\" devid=\"xxxxxx\" logid=\"0000000013\" type=\"traffic\" subtype=\" forward\" level=\"notice\" vd=\"root\" eventtime=xxxxtz=\"-0300\" srcip=xxxxxxxsrcport=60748 srcintf=\"lan5\" srcintfrole=\"lan\" dstip=8.8. 8.8 dstport=53 dstintf=\"wan2\" dstintfrole=\"wan\" dstinetsvc=\"Google-DNS\" sessionid=10438009 proto=17 action=\"accept\" policyid=47 policytype=\"policy\" poluuid=\"bfbf128e-8853-51e9-4dba-00103de9d520\" dstcountry=\"United States\" srccountry=\"Reserved\" trandisp=\"snat\" transip=xxxxxxtransport=60748 duration=181 sentbyte=104 rcvdbyte= 259 sentpkt=1 rcvdpkt=1 vwlid=0 appcat=\"unscanned\" mastersrcmac=\"68:4f:64:xx:x\" sxrcmac=\"68:xx:64:xx:9c:b5\" srcserver =0"
    ],
    "metadata.type.security": [
      "_doc"
    ],
    "metadata.type.keyword": [
      "_doc"
    ],
    "metadata._

**Comment by dougburks** [comment, marked as answer]
> I have a Fortigate Firewall that was configured to send the logs through the SYSLOG port, the logs appear inside Kibana but do not appear in a structured way, how can I format these logs to improve the visualization?

From https://docs.securityonion.net/en/2.3/syslog.html#syslog:
_If you want to send syslog from other devices, you should check to see if the device has an existing [Filebeat](https://docs.securityonion.net/en/2.3/filebeat.html#filebeat) module at https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-modules.html. If so, using the [Filebeat](https://docs.securityonion.net/en/2.3/filebeat.html#filebeat) module should provide some parsing by default._

Following the https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-modules.html link, included in the list is a Fortinet module:
https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-fortinet.html

> Is there a possibility to send an email notification, when something in the log of this fortigate reaches some pre-configured metric?

https://docs.securityonion.net/en/2.3/elastalert.html
